src/fbrs/controller/AddNewUserController.java

```diff
@@ -1,24 +1,121 @@
+                }
+                id = model.addUser(market.getId(), this.nameTextField.getText(), this.phoneTextField.getText(),
+                        User.getUserIDTYPE(user_type.getValue()));
+
+                message = "تمت إضافة التاجر لسوق " + this.marketComboBox.getValue() + " بنجاح";
+
+            } else if (user_type.getValue().equals("صياد لنش") || user_type.getValue().equals("صياد حسكة")) {
+                id = model.addUser(-1, this.nameTextField.getText(), this.phoneTextField.getText(),
+                        User.getUserIDTYPE(user_type.getValue()));
+                message = "تمت إضافة " + user_type.getValue() + " بنجاح";
+            } else {
+                return;
+            }
+        }
+        if (id == -1) {
+            message = "فشلت علمية الإضافة";
+            alert.setHeaderText(message);
+            alert.setContentText("يرجى التأكد من إدخال البيانات بشكل صحيح وعدم إستخدام إسم مستخدم موجود بالفعل");
+            alert.showAndWait();
+        } else {
+            alert.setAlertType(Alert.AlertType.INFORMATION);
+            alert.setTitle(message);
+            alert.setHeaderText("الاسم :" + this.nameTextField.getText());
+            alert.setContentText("الرقم التعريفي = " + id);
+            alert.showAndWait();
+            support.setErrorDecorationEnabled(false); // we don't want errors to bother us for now.
+            reset();
+        }
+    }
+
+    private void reset() {
+        nameTextField.setText("");
+        phoneTextField.setText("");
+        marketComboBox.setValue(null);
+        user_type.setValue(user_type.getPromptText());
     }
 
     @Override
     public void initialize(URL location, ResourceBundle resources) {
+        userTypeMap.put(1, "تاجر");
+        userTypeMap.put(5, "صياد لنش");
+        userTypeMap.put(6, "صياد حسكة");
+
+        List<Market> markets = model.getAllMarkets();
+        for (int i = 0; i < markets.size(); i++)
+            marketComboBox.getItems().add(markets.get(i));
+
+        user_type.getItems().addAll("تاجر", "صياد لنش", "صياد حسكة");
+        support.setErrorDecorationEnabled(false); // we don't want errors to bother us for now.
+        support.registerValidator(nameTextField, Validator.createEmptyValidator("يجب ادخال اسم المستخدم", Severity.ERROR));
+        support.registerValidator(user_type, Validator.createEmptyValidator("يجب إختيار نوع المستخدم", Severity.ERROR));
+
+    }
+
+    public void onSelectUserType() {
+        if (user_type.getValue().equals("صياد لنش") || user_type.getValue().equals("صياد حسكة")) {
+            marketComboBox.setVisible(false);
+            marketLabel.setVisible(false);
+            final Validator<String> validator = (final Control control, final String value) -> {
+                final boolean condition = true;
+                return ValidationResult.fromMessageIf(control, "not a number", Severity.ERROR, condition);
+            };
+            support.registerValidator(marketComboBox, validator);
```

Comment:
What is this?!
You can use RegexValidator instead
`support.registerValidator(numberField, Validator.createRegexValidator("must be digits only!", "\\d*", Severity.ERROR));`

Reply:
If a fisherman chooses, the market selection does not matter.
then reset Validator to send true always.
final boolean condition = true;